pyteal/compiler/optimizer/optimizer.py

```diff
@@ -24,10 +25,25 @@ class OptimizeOptions:
             that have no load dependencies elsewhere.
     """
 
-    def __init__(self, *, scratch_slots: bool = False):
-        self.scratch_slots = scratch_slots
+    def __init__(self, *, scratch_slots: bool = False, frame_pointers: bool = False):
+        self.scratch_slots: Final[bool] = scratch_slots
+        self.frame_pointers: Final[bool] = frame_pointers
         self._skip_slots: Set[ScratchSlot] = set()
 
+    @classmethod
+    def maximize(cls) -> "OptimizeOptions":
+        return cls(scratch_slots=True, frame_pointers=True)
+
+    @classmethod
+    def default_for_version(cls, version: int) -> "OptimizeOptions":
+        """TODO: consider maximizing for default starting with Teal v. 9"""
+        from pyteal.compiler.compiler import FRAME_POINTER_VERSION
+
+        if version < FRAME_POINTER_VERSION:
+            return cls(scratch_slots=False, frame_pointers=False)
```

Comment:
Is it too late to make the default behavior for v8 use frame pointers? If so, then I'm afraid that I'll have to default `frame_pointers = False` in the case of `version $\geq$ FRAME_POINTER_VERSION`.

Also, should I already aggressively force `maximize` for v9 and higher?

Reply:
It might be okay to make default v8 use of frame pointer, haven't thought through the consequences, simply feeling okay with me.

I am slightly against `maximize` for v9 and higher, for it might make debugging somehow harder?

Reply:
Default use of frame pointers is v8+ code is good with me.

I do not think we should default to `maximize` for any version at the moment. The risk of backwards incompatibility when new optimizations are introduced is too great to make this the default behavior in my opinion.

Reply:
I've removed `maximize()`. However, please note that as the PR currently stands, I'm actually defaulting to `optimize_scratch_slots() == True` starting with v9. This is open to discussion.